src/jobs/taskcluster_graph.js

```diff
@@ -29,6 +36,32 @@ function parseUrl(url) {
   };
 }
 
+/**
+Fetch a task graph from a url (retires included...)
+*/
+async function fetchGraph(job, url) {
+  assert(url, 'url is required');
+  job.log(`fetching graph ${url}`);
+  let currentRetry = 0;
+  while (currentRetry++ < GRAPH_RETIRES) {
+    try {
```

Comment:
yeah +1 : I think we are doing this pretty frequently at minimum I will extract this to a util in mozilla-tc